src/app/components/BookFilters/Filter.jsx

```diff
@@ -1,60 +1,82 @@
 import React from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import {
   CheckSoloIcon,
   DotsIcon,
 } from 'dgx-svg-icons';
 
+const ANIMATION_TIMEOUT = 400;
+
 class Filter extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      selected: false,
-      buttonActive: false,
-    };
+    this.state = { loading: false };
     this.onClick = this.onClick.bind(this);
   }
 
-  onClick() {
-    const filterId = this.props.filter.toLowerCase().split(' ').join('-');
-    this.props.onClick(filterId, !this.state.selected);
-    this.setState({
-      selected: !this.state.selected,
-      buttonActive: true,
-    });
-
+  componentDidMount() {
+    // Since the list of filters get re-rendered, when the a new list is generated and it's
+    // longer than the previous list, then we need to focus on the selected/deselected filter
+    // once all the filters are mounted.
     setTimeout(() => {
-      this.setState({ buttonActive: false });
-    }, 1000);
+      if (this.props.filter.id === this.props.focusId) {
+        ReactDOM.findDOMNode(this.refs[this.props.filter.id]).focus();
+      }
+    }, ANIMATION_TIMEOUT);
   }
 
-  render() {
+  componentWillReceiveProps(nextProps) {
     const {
-      selected,
-      buttonActive,
-    } = this.state;
-    const filter = this.props.filter;
-    const activeClass = selected ? 'active' : '';
-    const iconType = buttonActive ? <DotsIcon /> : <CheckSoloIcon />;
+      filter,
+      focusId,
+    } = nextProps;
+    // If the focusId, the filter that was JUST selected/deselected, matches the filter,
+    // then load the animation and remove it shortly after. Also focus on it whether it was
+    // selected or deselected for accessibility.
+    if (filter.id === focusId) {
+      this.setState({
```

Comment:
can probably be on one line since it updates a single state object property

Reply:
Updated.